model_inspector/inspect/any_model.py

```diff
@@ -28,16 +28,51 @@ class _Inspector:
     subclasses directly.
     """
 
-    def __init__(self, model: BaseEstimator, X: pd.DataFrame, y: pd.Series):
+    def __init__(
+        self,
+        # We aim to support estimators with `scikit-learn`-compatible
+        # interfaces from popular libraries such as `xgboost` and
+        # `catboost` even if they do not inherit from `BaseEstimator`,
+        # but I do not know of a simple way to include them in this type
+        # hint without adding those libraries as dependencies, which I
+        # would like to avoid if possible.
+        model: BaseEstimator,
+        X: pd.DataFrame,
+        y: pd.Series,
+    ):
         check_is_fitted(model)
         check_X_y(X, y)
-        if not isinstance(model, (DummyClassifier, DummyRegressor)):
-            model._check_n_features(X, reset=False)
+        self._check_cols(model, X)
 
         store_attr()
 
     __repr__ = basic_repr(["model"])
 
+    def _check_cols(self, model, X):
+        try:
+            if not model.feature_names_in_.equals(X.columns):
```

Comment:
Isn't equality of lists a bit too strong? I think it should be okay if the columns are permutes so maybe we could compare sets here instead?

Reply:
I think `sklearn` relies on the column order being the same for `.predict` as it was for `.fit`? If so, and maybe even if not, that seems to me like a reasonable thing to expect the user to handle. We could try to help out e.g. by reorder the columns at inspector creation time, but I suspect that approach would cause confusion about as often as it was helpful. What do you think?